lib/remap-functions/src/parse_grok.rs

```diff
@@ -110,7 +110,7 @@ impl Expression for ParseGrokFn {
     fn type_def(&self, state: &state::Compiler) -> TypeDef {
         self.value
             .type_def(state)
-            .fallible_unless(value::Kind::Bytes)
+            .into_fallible(true)
```

Comment:
I don't see why this one should be always fallible?

(although I did read somewhere we wanted to update this one to be fallible if it didn't match, which makes sense I think, but I don't think the implementation is updated yet?)

Reply:
> I don't see why this one should be always fallible?

Oh, good catch. Faulty assumption on my part, we should update this at that point.